app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -122,35 +123,125 @@ class MessagesController @Inject() (
-        } yield submitted).valueOr(
-          error => Status(error.code.statusCode)(Json.toJson(error))
-        )
+
+          customsOffice <- customOfficeExtractorService.extractCustomOffice(source.headOption.get, requestMessageType).asPresentation
+          size          <- calculateSize(source.lift(1).get)
+          submitted <-
+            if (config.eisSizeLimit >= size) viaEIS(messageType, customsOffice, source.lift(2).get)
+            else viaSDES(source.lift(3).get)
+        } yield submitted)
+          .valueOr(
+            error => Status(error.code.statusCode)(Json.toJson(error))
+          )
     }
   }
 
+  private def materializeOutgoingSource(source: Source[ByteString, _]): EitherT[Future, PresentationError, Seq[ByteString]] =
+    EitherT(
+      source
+        .runWith(Sink.seq)
+        .map(Right(_): Either[PresentationError, Seq[ByteString]])
+        .recover {
+          error =>
+            Left(PresentationError.internalServiceError(cause = Some(error)))
+        }
+    )
+
+  // Function to materialize the source into a Seq[ByteString] with error handling
+  private def materializeIncomingSource(source: Source[ByteString, _]): EitherT[Future, PresentationError, Seq[ByteString]] =
+    EitherT(
+      source
+        .via(eisMessageTransformers.unwrap)
+        .runWith(Sink.seq)
+        .map(Right(_): Either[PresentationError, Seq[ByteString]])
+        .recover {
+          case error: Throwable if error.toString.contains("IllegalStateException") || error.toString.contains("WFCException") =>
+            if (config.logIncoming) {
+              logger.error(
+                s"""Unable to process message from EIS -- bad request:
+               |
+               |Message: ${error.getMessage}
+               |
+               |Failed to transform XML""".stripMargin,
+                error
+              )
+            }
+            Left(PresentationError.badRequestError(error.getMessage))
+          case error: Throwable =>
+            if (config.logIncoming) {
+              logger.error(
+                s"""Unable to process message from EIS -- internal server error:
+               |
+               |Message: ${error.getMessage}
+               |
+               |Failed to transform """.stripMargin,
+                error
+              )
+            }
+            Left(PresentationError.internalServiceError(cause = Some(error)))
+        }
+    )
+
+  // Function to create a new source from the materialized sequence
+  private def createReusableSource(seq: Seq[ByteString]): Source[ByteString, _] = Source(seq.toList)
+
+  // Materialize the source and create multiple new sources with error handling
+  private def reUsableIncomingSource(request: Request[Source[ByteString, _]]): EitherT[Future, PresentationError, List[Source[ByteString, _]]] = for {
+    byteStringSeq <- materializeIncomingSource(request.body)
+  } yield List.fill(4)(createReusableSource(byteStringSeq))
```

Comment:
Rather than a magic number 4 and so we know why, can we declare something like
val numberOfSourceCopiesRequired = 4

List.fill(numberOfSourceCopiesRequired)